secrets/portworx-with-kubernetes-secrets.md

```diff
@@ -0,0 +1,130 @@
+>**Note:**<br/>Supported from PX Enterprise 1.4 onwards
+
+### Configuring Kubernetes Secrets with Portworx
+
+#### New installation
+When generating the [Portworx Kubernetes spec file](https://install.portworx.com/), select `Kubernetes` from the "Secrets type" list. For more details on how to generate Portworx spec for Kubernetes, refer instructions on [Deploy Portworx on Kubernetes](/scheduler/kubernetes/install.html).
+
+#### Existing installation
+
+##### Permissions to access secrets
+Portworx stores credentials/secrets in a Kubernetes namespace called `portworx`. It needs permissions to access secrets under this namespace. If you have upgraded Portworx using the [recommended method](/scheduler/kubernetes/upgrade-1.3.html), then you will not have to create the namespace and roles given below. If the following objects are missing, then create it using `kubectl`:
+```yaml
+cat <<EOF | kubectl apply -f -
+
+## Namespace to store credentials
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: portworx
+---
+
+## Role to access secrets under portworx namespace only
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: px-role
+  namespace: portworx
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "list", "create", "update", "patch"]
+---
+
+## Allow portworx service account to access the secrets under the portworx namespace
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: px-role-binding
+  namespace: portworx
+subjects:
+- kind: ServiceAccount
+  name: px-account
+  namespace: kube-system
+roleRef:
+  kind: Role
+  name: px-role
+  apiGroup: rbac.authorization.k8s.io
+
+EOF
+```
+
+##### Update config.json
+After ensuring the `portworx` namespace and the required permissions are present, you will have to update the `/etc/pwx/config.json` to start using Kubernetes secrets by default. Add the `secret_type` and `cluster_secret_key` fields in the `secret` section to the `/etc/pwx/config.json` on each node in the cluster:
+```json
+{
+    "clusterid": "",
+    "secret": {
+        "secret_type": "k8s",
+        "cluster_secret_key": "cluster-wide-secret-key"
+    },
+    ...
+}
+```
+
+##### Edit the Portworx Daemonset
+You will have to edit the Portworx daemonset to use Kubernetes secrets, so that all the new Portworx nodes will start using Kubernetes secrets. You will not have to change the *config.json* for the new nodes if you edit the daemonset.
+```
+# kubectl edit daemonset portworx -n kube-system
+```
+Add the `"-secret_type", "k8s"` arguments to the `portworx` container in the daemonset. It should look something like this:
+```yaml
+  containers:
+  - args:
+    - -c
+    - testclusterid
+    - -s
+    - /dev/sdb
+    - -x
+    - kubernetes
+    - -secret_type
+    - k8s
+    name: portworx
+    image: portworx/oci-monitor:1.4.0
```

Comment:
remove the image line. Will get stale over time and would need constant updating.